### PR TITLE
Wallet Connection

### PR DESCRIPTION
**Pull request by Guilherme775**
#36 

This PR adds and exports a function to connect to Nami or Gero wallet.

Quick video connecting in the front:

https://user-images.githubusercontent.com/55771765/183759438-252eef05-47e9-4315-a8ff-0bf1d518362e.mp4

### CODE REVIEW

**Comment by Guilherme775** [comment]
@samuelWilliams99 I opened the PR just here adding/exporting the function, should I open a PR in the nft-marketplace doing the integration?

File: index.d.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export function callMarketPlaceListNft(config: Config):
  */
 export function callMarketPlaceFetchNft(config: Config, args: FetchNftArgs):
     Promise<NftListing?>
-export function connectWallet(): Promise<any>
+export function connectWallet(walletOption: string): Promise<any>
```
Comment (rynoV) [inline index.d.ts:12]: Let's make a typescript type here like `type WalletOption = "Nami" | "Gero"`

File: src/Seabug/CallContract.purs [hunk 2]
```diff
@@ -67,6 +69,16 @@ import Serialization.Hash
...
+import Wallet (Wallet, mkNamiWalletAff, mkGeroWalletAff)
+
+callConnectWallet
+  :: forall (r :: Row Type)
+   . String
+  -> Effect (Promise Wallet)
+callConnectWallet walletOption = case walletOption of
+  "Nami" -> Promise.fromAff mkNamiWalletAff
+  "Gero" -> Promise.fromAff mkGeroWalletAff
+  _ -> throwError <<< error $ "Unsupported wallet"
```
Comment (rynoV) [inline src/Seabug/CallContract.purs:81]: Let's include `walletOption` in the error message

**Review by rynoV — DISMISSED** [review]
Just some small things, but otherwise lgtm

**Comment by samuelWilliams99** [comment]
Given the "no nfts found", @rynoV, could you run this with your testnet node? I'd like to know if the nfts will show up without a wallet connection

File: src/Seabug/Seabug.purs [hunk 3]
```diff
@@ -36,4 +37,4 @@ mint = callMint
   , name: "Piaggio Ape"
   , description: "Seabug Testing"
   , price: BigInt.fromInt (12 * 1000000)
-  }
+  }
```
Comment (samuelWilliams99) [inline src/Seabug/Seabug.purs:40]: Lets keep newlines at the end of files

**Comment by rynoV** [comment]
> I opened the PR just here adding/exporting the function, should I open a PR in the nft-marketplace doing the integration?
@Guilherme775 Yeah if you have some code changes there just open a PR, even if it's just a draft

**Comment by rynoV** [comment]
> I'd like to know if the nfts will show up without a wallet connection

@samuelWilliams99 It's the same as before, where if nami isn't connected the website prompts the user to connect nami before loading anything

**Comment by samuelWilliams99** [comment]
Okay - part of this PR needs to be to remove that behaviour, and allow the contracts to query nfts without knowing own pubkeyhash.
@Guilherme775 could you look into this? I'd start by verifying with the CTL team whether or not this is even possible

**Comment by Guilherme775** [comment]
@samuelWilliams99 just remove Nami from walletSpec pretty easy, now the NFTs are listed without connecting the wallet

**Comment by samuelWilliams99** [comment]
> @samuelWilliams99 just remove Nami from walletSpec pretty easy, now the NFTs are listed without connecting the wallet
Right - do buy transactions still work after you link your wallet?

**Review by samuelWilliams99 — APPROVED** [review]

**Comment by Guilherme775** [comment]
> > @samuelWilliams99 just remove Nami from walletSpec pretty easy, now the NFTs are listed without connecting the wallet
> 
> Right - do buy transactions still work after you link your wallet? Does it always use the most recently linked wallet?

Yep, seems to be working fine

**Comment by Guilherme775** [comment]
After merge this, would be good review/merge this one: https://github.com/mlabs-haskell/nft-marketplace/pull/227

**Pull request merged**